ui/app/models/pki/key.js

```diff
@@ -1,17 +1,40 @@
 import Model, { attr } from '@ember-data/model';
 import { inject as service } from '@ember/service';
 import { withFormFields } from 'vault/decorators/model-form-fields';
+import { withModelValidations } from 'vault/decorators/model-validations';
 
-@withFormFields(['keyId', 'keyName', 'keyType', 'keyBits'])
+const validations = {
+  type: [{ type: 'presence', message: 'Type is required.' }],
+};
+const displayFields = ['keyId', 'keyName', 'keyType', 'keyBits'];
+const formFieldGroups = [{ default: ['keyName', 'type'] }, { 'Key parameters': ['keyType', 'keyBits'] }];
+@withModelValidations(validations)
+@withFormFields(displayFields, formFieldGroups)
 export default class PkiKeyModel extends Model {
   @service secretMountPath;
 
-  @attr('boolean') isDefault;
-  @attr('string', { possibleValues: ['internal', 'external'] }) type;
   @attr('string', { detailsLabel: 'Key ID' }) keyId;
-  @attr('string') keyName;
-  @attr('string') keyType;
-  @attr('string') keyBits;
+  @attr('string', { subText: 'Optional, human-readable name for this key.' }) keyName;
+  @attr('string') privateKey;
+  @attr('string', {
+    noDefault: true,
+    possibleValues: ['internal', 'exported'],
```

Comment:
Just checking that these values changed on purpose from `internal, external` to `internal, exported`

Reply:
Yes! For some reason I keep typing it everywhere as `external` (i.e. in curl requests...) but it's `exported` 😶‍🌫️